rotkehlchen/tests/unit/decoders/test_yearn.py

```diff
@@ -191,8 +193,227 @@ def test_withdraw_yearn_v3(
+@pytest.mark.parametrize('use_clean_caching_directory', [True])
+def test_deposit_yearn_v2(
+        ethereum_inquirer: 'EthereumInquirer',
+        ethereum_accounts: list['ChecksumEvmAddress'],
+) -> None:
+    tx_hash = deserialize_evm_tx_hash('0x7e6493bb2e0c09b9986ab43a562409fd2cf99caca3d1b65fa4b0d46ef5588081')  # noqa: E501
+    get_or_create_evm_token(
+        userdb=ethereum_inquirer.database,
+        evm_address=string_to_evm_address('0xcC2EFb8bEdB6eD69ADeE0c3762470c38D4730C50'),
+        chain_id=ChainID.ETHEREUM,
+        token_kind=EvmTokenKind.ERC20,
+        symbol='yvCurve-crvDOLA-f',
+        name='Curve crvDOLA Factory yVault',
+        decimals=18,
+        protocol=YEARN_VAULTS_V2_PROTOCOL,
+        started=Timestamp(1635529757),
+        underlying_tokens=[UnderlyingToken(
+            address=string_to_evm_address('0xef484de8C07B6e2d732A92B5F78e81B38f99f95E'),
+            token_kind=EvmTokenKind.ERC20,
+            weight=ONE,
+        )],
+    )
+    events, _ = get_decoded_events_of_transaction(evm_inquirer=ethereum_inquirer, tx_hash=tx_hash)
+    timestamp, user_address, gas_amount, deposit_amount, receive_amount, approve_amount = TimestampMS(1729145687000), ethereum_accounts[0], '0.001446241576196176', '38541.366598671832692528', '38514.207134567395983686', '57896044618658097711785492504343953926634992332820281981187.425405284732127439'  # noqa: E501
+    assert events == [
+        EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal(gas_amount)),
+            location_label=user_address,
+            notes=f'Burned {gas_amount} ETH for gas',
+            counterparty=CPT_GAS,
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=364,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.DEPOSIT,
+            event_subtype=HistoryEventSubType.DEPOSIT_ASSET,
+            asset=Asset('eip155:1/erc20:0xef484de8C07B6e2d732A92B5F78e81B38f99f95E'),
+            balance=Balance(amount=FVal(deposit_amount)),
+            location_label=user_address,
+            notes=f'Deposit {deposit_amount} crvDOLA in yearn-v2 vault',
+            counterparty=CPT_YEARN_V2,
+            address=YEARN_PARTNER_TRACKER,
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=365,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.APPROVE,
+            asset=Asset('eip155:1/erc20:0xef484de8C07B6e2d732A92B5F78e81B38f99f95E'),
+            balance=Balance(amount=FVal(approve_amount)),
+            location_label=user_address,
```

Comment:
The order here is off. Deposit shoudd be first, followed by the token receival. OUT, then IN. Approval should be pushed out. You need to use maybe_reshuffle_Events in the decoder